### PR TITLE
Update ci.yml and CMakePresets to enable gcc-{12, 13} and clang-19

### LINKED ISSUE #17: [MEDIUM] Enable latest versions of compilers on main
[MEDIUM] Enable latest versions of compilers on main

- [ ] Enable gcc-15
- [x] Enable gcc-14
- [x] Enable gcc-13
- [x] Enable gcc-12
- [ ] Enable clang-20
- [x] Enable clang-19
- [x] Enable clang-18
- [x] Enable clang-17
- [ ] Enable clang-16
- [ ] TBD msvc

### LINKED ISSUE #18: [LOW] Re-enable ASAN run CI/clang-19
ASAN run was temporary disabled for clang-19 on CI. Similar motivation with https://github.com/beman-project/optional26/issues/32

TODOs:
- [ ] Enable ASAN run with Clang-19 on CI
- [ ] Use `gtest_add_tests`

### PR DESCRIPTION
**Pull request by RaduNichita**
The PR imports the `ci.yml` and `CMakePresets` from [optional26 repository](https://github.com/beman-project/optional26):

- [x] Enable ASAN run with Clang-19 on CI

- [x] Enable gcc-14
- [x] Enable gcc-13
- [x] Enable gcc-12
- [x] Enable clang-19
- [x] Enable clang-18
- [x] Enable clang-17

Partially Fixes #18  and partially fixes #17

### CODE REVIEW

File: CMakePresets.json [hunk 1]
```diff
@@ -31,12 +31,33 @@
       "description": "Build with GCC 14 compilers",
       "toolchainFile": "${sourceDir}/etc/gcc-14-toolchain.cmake"
     },
+    {
+      "name": "gcc-13",
+      "inherits": "common",
+      "displayName": "GCC 13",
```
Comment (neatudarius) [inline CMakePresets.json:37]: Please also add gcc-12

**Review by neatudarius — COMMENTED** [review]
Welcome to Beman and congratulations for your first contribution!

File: CMakePresets.json (hunk 1 above)
Comment (RaduNichita) [reply CMakePresets.json:37]: done

**Review by neatudarius — APPROVED** [review]
LGTM 👍 

I also tested all preset combinations on my local Ubuntu. 

clang-{17, 18, 19} and gcc-{12, 13, 14} work on local + CI setup.


Thanks again @RaduNichita and welcome to our community! Looking forward for other contributions!

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,75 +1,42 @@
-name: Test
-
+name: CI Tests
```
[the remark this answers is not in the document]
Comment (RaduNichita) [reply .github/workflows/ci.yml:1]: updated description

**Pull request merged**